src/Avalonia.Controls/Flyouts/FlyoutBase.cs

```diff
@@ -0,0 +1,466 @@
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.Bottom;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Bottom;
+                    break;
+
+                case FlyoutPlacementMode.BottomEdgeAlignedLeft:
+                    _popup.PlacementRect = new Rect(0, trgtBnds.Height - 1, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.BottomRight;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Bottom;
+                    break;
+
+                case FlyoutPlacementMode.BottomEdgeAlignedRight:
+                    _popup.PlacementRect = new Rect(trgtBnds.Width - 1, trgtBnds.Height - 1, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.BottomLeft;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Bottom;
+                    break;
+
+                case FlyoutPlacementMode.LeftEdgeAlignedTop:
+                    _popup.PlacementRect = new Rect(0, 0, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.BottomLeft;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Left;
+                    break;
+
+                case FlyoutPlacementMode.Left: //Left & centered
+                    _popup.PlacementRect = new Rect(0, 0, 1, trgtBnds.Height);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.Left;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Left;
+                    break;
+
+                case FlyoutPlacementMode.LeftEdgeAlignedBottom:
+                    _popup.PlacementRect = new Rect(0, trgtBnds.Height - 1, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.TopLeft;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.BottomLeft;
+                    break;
+
+                case FlyoutPlacementMode.Full:
+                    //Not sure how the get this to work
+                    //Popup should display at max size in the middle of the VisualRoot/Window of the Target
+                    throw new NotSupportedException("FlyoutPlacementMode.Full is not supported at this time");
+                //break;
+
+                //includes Auto (not sure what determines that)...
+                default:
+                    //This is just FlyoutPlacementMode.Top behavior (above & centered)
+                    _popup.PlacementRect = new Rect(-sz.Width / 2, 0, sz.Width, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.Top;
+                    break;
+            }
+        }
+
+        private static void OnContextFlyoutPropertyChanged(AvaloniaPropertyChangedEventArgs args)
+        {
+            if (args.Sender is Control c)
+            {
+                if (args.OldValue is FlyoutBase)
```

Comment:
Keyboard "Menu" key should be also supported.

Reply:
Is there a mapping to the "menu" key in the `Key` enum? It seems like `Key.Apps` is the match if you handle OnKeyDown, but I'm not sure if this is correct. `ContextMenu` didn't handle this, at least I can't find where and the menu key doesn't appear to do anything. Shift+F10 is an alternate way in Windows too, but I don't know about Mac/Linux

Reply:
> Shift+F10

Ah, yes, forgot about this one. It should be implemented too.
And yes, it seems Avalonia doesn't have any of them.

In Avalonia we have "platform hotkeys" implementation for different platforms.
I believe "ContextMenu" should be also included in this list, so it can be configured per platform as well.
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Input/Platform/PlatformHotkeyConfiguration.cs
Later specific keydown event args can be matched with these gestures like it was done in TextBox class
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/TextBox.cs#L602

Reply:
I'm thinking it may be best to hold off for now then (but I agree this should be added), and I'll defer to someone who knows a little more about the cross platform key gestures - I don't want to wire something up incorrectly. Since ContextMenu didn't already do this, the lack of it shouldn't be noticeable for now.